### PR TITLE
 Merge the .Net Native and CoreCLR implementations of EventSource. Th…

### PR DESCRIPTION
**Pull request by davmason**
…is replaces a lot of the previous ifdefs in the ProjectN impelmentation with a partial class that is cleaner and easier to understand.

 # This is a combination of 5 commits.
Merge the .Net Native and CoreCLR implementations of EventSource

Merge the CoreCLR refactoring in to the ProjectN folder. After this change the project should be much easier to merge and all bug fixes are up to date.

Refactor the interop in ProjectN so it can work without ifdefs.

Move methods that use EventActivityIDControl to EventSource_ProjectN.c

Add formatting to the debugger message that EventSource prints when an internal error happens.

Port remaining fixes to N from CoreCLR.

### CODE REVIEW

**Comment by davmason** [comment]
@vancem this makes the source identical between CoreCLR and CoreFX.

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Release Build and Test please
Test Innerloop Ubuntu Debug Build and Test please

File: src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/ActivityTracker.cs [hunk 1]
```diff
@@ -53,21 +57,31 @@ internal class ActivityTracker
         public void OnStart(string providerName, string activityName, int task, ref Guid activityId, ref Guid relatedActivityId, EventActivityOptions options)
         {
             if (m_current == null)        // We are not enabled
-                return;
+            {
+                // We  used to rely on the TPL provider turning us on, but that has the disadvantage that you don't get Start-Stop tracking
+                // until you use Tasks for the first time (which you may never do).   Thus we change it to pull rather tan push for whether
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/ActivityTracker.cs]: Nit: "tan" => "than"

**Comment by vancem** [comment]
LGTM

**Pull request merged**